code-studio/build-commands.js

```diff
@@ -49,10 +49,14 @@ exports.browserify = function (config) {
   var command = (shouldWatch ? 'watchify -v' : 'browserify') +
     (shouldMinify ? '' : ' --debug');
 
+  var reactifyStep = '-t reactify';
+
+  var globifyStep = '-t require-globify';
+
   var factorStep = '';
   if (shouldFactor) {
     factorStep = "-p [ factor-bundle -o '" + (shouldMinify ? 'uglifyjs ' : '') +
-      '>' + buildPath + "`basename $FILE .js`" + extension + "']";
+      '>' + buildPath + "`dirname $(realpath --relative-to " + srcPath + " $FILE)`/`basename -s .js -s .jsx $FILE`" + extension + "']";
```

Comment:
I agree with that sentiment. I was struggling with this some too when I was wanting to do renames. I wonder if this becomes easier if we switch to partition-bundle, which appears to be able to take a mapping config https://github.com/arian/partition-bundle#configuration-file